server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportGetAllocationStatsAction.java

```diff
@@ -110,23 +113,35 @@ public ActionRequestValidationException validate() {
     public static class Response extends ActionResponse {
 
         private final Map<String, NodeAllocationStats> nodeAllocationStats;
+        private DiskThresholdSettings diskThresholdSettings = null;
 
-        public Response(Map<String, NodeAllocationStats> nodeAllocationStats) {
+        public Response(Map<String, NodeAllocationStats> nodeAllocationStats, DiskThresholdSettings diskThresholdSettings) {
             this.nodeAllocationStats = nodeAllocationStats;
+            this.diskThresholdSettings = diskThresholdSettings;
         }
 
         public Response(StreamInput in) throws IOException {
             super(in);
             this.nodeAllocationStats = in.readImmutableMap(StreamInput::readString, NodeAllocationStats::new);
+            if (in.getTransportVersion().onOrAfter(TransportVersions.WATERMARK_THRESHOLDS_STATS)) {
+                this.diskThresholdSettings = DiskThresholdSettings.readFrom(in);
+            }
         }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             out.writeMap(nodeAllocationStats, StreamOutput::writeString, StreamOutput::writeWriteable);
+            if (out.getTransportVersion().onOrAfter(TransportVersions.WATERMARK_THRESHOLDS_STATS)) {
+                diskThresholdSettings.writeTo(out);
+            }
```

Comment:
If we check `clusterService.state().getMinTransportVersion()` against `TransportVersions.WATERMARK_THRESHOLDS_STATS` as I suggested above then we can assert that `diskThresholdSettings == null` in the `else` branch here, good for future readers.